### PR TITLE
Ts/calendar day

### PR DESCRIPTION
**Pull request by lidord-wix**
Finish calendar day migration to TS.
Return the dot propTypes (it was necessary).

### CODE REVIEW

File: src/calendar/day/dot/index.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import PropTypes from 'prop-types';
```
Comment (Inbal-Tish) [inline src/calendar/day/dot/index.tsx:1]: Why are we using prop-types here?

File: src/calendar/day/index.tsx [hunk 2]
```diff
@@ -37,7 +47,7 @@ export default class Day extends Component {
     ]);
   }
 
-  getMarkingLabel(marking) {
+  getMarkingLabel(marking: any) {
```
Comment (Inbal-Tish) [inline src/calendar/day/index.tsx:50]: Why not object?

File: src/calendar/day/index.tsx [hunk 3]
```diff
@@ -68,8 +78,8 @@ export default class Day extends Component {
   }
 
   getAccessibilityLabel = memoize((day, marking, isToday) => {
-    const today = XDate.locales[XDate.defaultLocale].today;
-    const formatAccessibilityLabel = XDate.locales[XDate.defaultLocale].formatAccessibilityLabel;
+    const today = _.get(XDate, 'locales.[XDate.defaultLocale].today');
+    const formatAccessibilityLabel = _.get(XDate, 'locales[XDate.defaultLocale].formatAccessibilityLabel');
```
Comment (Inbal-Tish) [inline src/calendar/day/index.tsx:82]: Did you test these changes? I see that the first one includes `locales**.**[XDate.defaultLocale]` and the other one `locales[XDate.defaultLocale]`...

File: src/calendar/day/dot/index.tsx (hunk 1 above)
Comment (Inbal-Tish) [reply src/calendar/day/dot/index.tsx:1]: `marking` is using Dot props. We should include Marking component in this PR...

Comment (lidord-wix) [reply src/calendar/day/dot/index.tsx:1]: The `extractComponentProps` function (src/component-updater.js) is based on the prop-types, when removing it the unit test are failling. So we decided to keep it until we'll make the relevant adjustment..

File: src/calendar/day/index.tsx (hunk 2 above)
Comment (lidord-wix) [reply src/calendar/day/index.tsx:50]: Actually setting it to `object` is almost like setting it to `any` if we don't write the object keys.
And I don't really know the possible keys for `marking`

File: src/calendar/day/index.tsx (hunk 3 above)
Comment (lidord-wix) [reply src/calendar/day/index.tsx:82]: fixed :)

File: src/calendar/day/dot/index.tsx (hunk 1 above)
Comment (Inbal-Tish) [reply src/calendar/day/dot/index.tsx:1]: Ok

File: src/calendar/day/index.tsx (hunk 2 above)
Comment (Inbal-Tish) [reply src/calendar/day/index.tsx:50]: I disagree. 'object' is more restrictive than 'any' type and indicates that the value is a non-primitive.

File: src/calendar/day/index.tsx [hunk 4]
```diff
@@ -4,28 +4,39 @@ import XDate from 'xdate';
...
+// @ts-expect-error
 import {xdateToData} from '../../interface';
+// @ts-expect-error
 import {SELECT_DATE_SLOT} from '../../testIDs';
-import BasicDay from './basic';
+import BasicDay, {BasicDayProps} from './basic';
 import PeriodDay from './period';
+import {MarkingProps} from './marking';
+
+const basicDayPropsTypes = _.omit(BasicDay.propTypes, 'date');
```
Comment (Inbal-Tish) [inline src/calendar/day/index.tsx:19]: Why the change of name if you're not using `BasicDayProps` to get the 'date'?

Comment (lidord-wix) [reply src/calendar/day/index.tsx:19]: I found it confusing to have both `basicDayProps` and `BasicDayProps`

Comment (Inbal-Tish) [reply src/calendar/day/index.tsx:19]: I understand but why are you still using `BasicDay.propTypes`?

Comment (lidord-wix) [reply src/calendar/day/index.tsx:19]: This is for the Day propTypes - as I mentioned, we decided to keep all the propTypes for now...

Comment (Inbal-Tish) [reply src/calendar/day/index.tsx:19]: I understand why you're keeping the propTypes, I'm not sure why you're using it here...

Comment (lidord-wix) [reply src/calendar/day/index.tsx:19]: The Day propTypes is using the BasicDay.propTypes (as before) and the DayProps is using the BasicDayProps.

**Review by Inbal-Tish — APPROVED** [review]
